data_model/src/transaction.rs

```diff
@@ -342,28 +347,64 @@ impl Transaction {
         self.payload.nonce = Some(nonce);
         self
     }
+}
 
-    /// Sign transaction with the provided key pair.
-    ///
-    /// # Errors
-    ///
-    /// Fails if signature creation fails
-    #[cfg(feature = "std")]
-    pub fn sign(
+#[cfg(feature = "std")]
+impl Sign for Transaction {
+    fn sign(
+        self,
+        key_pair: iroha_crypto::KeyPair,
+    ) -> Result<SignedTransaction, iroha_crypto::Error> {
+        let signature = SignatureOf::new(key_pair, &self.payload)?;
+        let signatures = btree_set::BTreeSet::from([signature]);
+
+        Ok(SignedTransaction {
+            payload: self.payload,
+            signatures,
+        })
+    }
+}
+
+impl Txn for Transaction {
+    type HashOf = SignedTransaction;
+
+    #[inline]
+    fn payload(&self) -> &Payload {
+        &self.payload
+    }
+}
+
+/// Structure that represents the second state of the transaction after receiving at least one signature.
+///
+/// `Iroha` and its clients use [`SignedTransaction`] to send transactions over the network.
+/// Before further processing,
+/// the transaction must be accepted by the `Iroha` peer: verify the signatures and check the limits.
```

Comment:
```suggestion
/// After a transaction is signed and before it can be processed any further,
/// the transaction must be accepted by the `Iroha` peer. The peer verifies the signatures and checks the limits.
```